src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTask.cs

```diff
@@ -570,19 +538,51 @@ private List<string> GetFilesForRestore(string backupLocation)
             }
         }
 
-        private Stream GetSnapshotInputStream(Stream fileStream, string database)
+        private void RestoreFromLastFile(Action<IOperationProgress> onProgress, DocumentDatabase database, string snapshotFile, DocumentsOperationContext context, bool onlySnapshot = false)
```

Comment:
You are reading the same file twice.
First reading the subscriptions from it and and then restoring the documents.

Reply:
done